apiserver/modelconfig/modelconfig.go

```diff
@@ -31,7 +31,7 @@ type ModelConfigAPI struct {
 
 // NewModelConfigAPI creates a new instance of the ModelConfig Facade.
 func NewModelConfigAPI(backend Backend, authorizer facade.Authorizer) (*ModelConfigAPI, error) {
-	if !authorizer.AuthClient() {
+	if !authorizer.AuthClient() && !authorizer.AuthUnitAgent() {
```

Comment:
This doesn't look correct. The UnitAgent already has its own API Facade to use. Instead of trying to use a different Facade, you should be adding the function that you need on its Facade. (Uniter).

ModelConfig is the Client (aka User) API, it should not be used by Agents.

Even further, you should not be asking for all of ModelConfig, you should be calling a function that only returns the details you actually need.
It might be *backed* by ModelConfig (that is where it gets the data), but it should not be asking for everything-that-could-be-configured-ever. Otherwise when we refactor where data is stored, it requires auditing for "you're asking for a giant map, what bits do you actually depend on".